### PR TITLE
support BIP68

### PR DESCRIPTION
**Pull request by matiu**
* Add BIP68 support to BTC and BCH bitcore-lib.
 * add /v4/txproposals POST endpoint to create new TX v2 for BTC/BCH.
 * add /v2/txproposals/<id>/signatures to post signatures to TX v2 for BTC/BCH.
 * throws error (need_upgrade) if /v1/txproposals/<id>/signatures tries to sign a TX v2

### CODE REVIEW

File: packages/bitcore-lib-cash/lib/transaction/input/input.js [hunk 1]
```diff
@@ -201,4 +210,67 @@ Input.prototype._estimateSize = function() {
...
+ */
+Input.prototype.getLockTime = function() {
+  if (this.sequenceNumber & SEQUENCE_LOCKTIME_DISABLE_FLAG) {
+    return null;
+  }
+
+  if (this.sequenceNumber & SEQUENCE_LOCKTIME_TYPE_FLAG) {
+    var seconds = SEQUENCE_LOCKTIME_GRANULARITY * (this.sequenceNumber & SEQUENCE_LOCKTIME_MASK);
+    return seconds;
+    return s;
```
Comment (nitsujlangston) [inline packages/bitcore-lib-cash/lib/transaction/input/input.js:266]: remove this line

File: packages/bitcore-lib/lib/transaction/input/input.js [hunk 2]
```diff
@@ -216,4 +223,67 @@ Input.prototype._estimateSize = function() {
...
+ */
+Input.prototype.getLockTime = function() {
+  if (this.sequenceNumber & SEQUENCE_LOCKTIME_DISABLE_FLAG) {
+    return null;
+  }
+
+  if (this.sequenceNumber & SEQUENCE_LOCKTIME_TYPE_FLAG) {
+    var seconds = SEQUENCE_LOCKTIME_GRANULARITY * (this.sequenceNumber & SEQUENCE_LOCKTIME_MASK);
+    return seconds;
+    return s;
```
Comment (nitsujlangston) [inline packages/bitcore-lib/lib/transaction/input/input.js:279]: remove this line

**Comment by cmgustavo** [comment]
Testing the PR in the the app I got this error creating a proposal:

Screen Shot 2020-04-05 at 21 19 21

Not sure why, because I created the tx v4 without issue:

```
ions: []
version: 4
createdOn: 1586132350
id: "afddfc97-87bf-4064-93cc-8ae601f5073e"
walletId: "f39c02bd-e1b4-419e-9b17-e0cdaea6a7d3"
creatorId: "13f14b2914e676a14e7308bffdaa479f779814d8d2af08ceadd85b13eaf48514"
coin: "btc"
network: "testnet"
``` 

I have BWC and BWS running and pointed to the app.

**Comment by matiu** [comment]
thanks for testing! 

That is because bitcore-lib(-cash) are also changed on this PR. To test it on the app you need to link them also.

**Pull request merged**